draft-ietf-httpbis-cache-latest.xml

```diff
@@ -589,15 +589,15 @@
    responses without a Vary header field), the cache will need to choose one to use.
    When a selecting header field has a known mechanism for doing so (e.g., qvalues on
    <x:ref>Accept</x:ref> and similar request header fields), that mechanism &MAY; be
-   used to select preferred responses; of the remainder, the most recent
+   used to select a preferred response. If such a mechanism is not available, or leads to equally preferred responses, the most recent
    response (as determined by the <x:ref>Date</x:ref> header field) is used, as
    per <xref target="constructing.responses.from.caches"/>.
 </t>
 <t>
    Note that in practice, some resources might send the Vary header field on
    responses inconsistently. When a cache has multiple responses for a
    target URI, and one or more omits the Vary header field, it &SHOULD; use the
-   most recent non-empty value available to select an appropriate response
+   most recently received non-empty Vary field value available to select an appropriate response
```

Comment:
I don't think this change is correct -- "recent" includes adjustment by Age, whereas most recently received means something else.

I also don't think the original requirement is correct, but it may reflect an old argument I lost long ago. Basically, the notion that an old Vary field is more relevant than a newer response without Vary is obviously wrong as soon as a resource stops varying.

Reply:
```suggestion
   most recent (see <xref target="age.calculations"/>) non-empty Vary field value available to select an appropriate response
```

Reply:
@royfielding See ^^

Reply:
FWIW: the change about recent *non-empty* was added in https://github.com/httpwg/http-core/commit/b9789cb51370caa6c9c73d9eb70806691b3285e1#diff-c2c4977337243f1b71ca2e3fd5b2e00dc95ebf32f8f63594e3b85eb21df30230 - I'm not sure whether this change was correct, but if we stick with it we may want to mention it in "changes from 7234".

Reply:
FWIW, the addition of

> 
"Note that in practice, some resources might send the Vary header field on responses inconsistently. When a cache has multiple responses for a given target URI, and one or more omits the Vary header field, it SHOULD use the most recent non-empty value available to select an appropriate response for the request."

happenen in draft 06. Isn't this a normative change that needs to be listed under "changes from RFC 7234"?

Reply:
I think it is flat out wrong and should be reverted. An empty Vary is just as valid as a full Vary and that shouldn't affect which valid stored response is chosen.